packages/dashboard/src/components/Wizard/SetupWizard.Base.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { ISetupWizardProps, ISetupWizardStyles, ISetupWizardStyleProps } from './SetupWizard.types';
+import { getSetupWizardStyles } from './SetupWizard.styles';
+import { classNamesFunction } from 'office-ui-fabric-react/lib/Utilities';
+import { Wizard } from './Wizard';
+import { getStepContentToShow, getCurrentStep } from './Wizard.utils';
+
+const getClassNames = classNamesFunction<ISetupWizardStyleProps, ISetupWizardStyles>();
+
+/** Component for Setup Wizard */
+export class SetupWizardBase extends React.Component<ISetupWizardProps, {}> {
+  constructor(props: ISetupWizardProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const classNames = getClassNames(getSetupWizardStyles!);
+    const stepContentToShow = getStepContentToShow(this.props.wizardProps);
+
+    return (
+      <div className={classNames.wizardContainer}>
+        {this._onRenderTitleSection()}
+        <Wizard {...this.props.wizardProps} contentToShow={stepContentToShow} />
+        {this._onRenderActionBar()}
+      </div>
+    );
+  }
+
+  // Get wizard title section
+  private _onRenderTitleSection = (): React.ReactNode => {
+    const classNames = getClassNames(this.props.styles!, { theme: this.props.theme! });
+    const step = getCurrentStep(this.props.wizardProps.steps);
+
+    return <div className={classNames.titleSection}>{step.titleElement}</div>;
+  };
+
+  // Get wizard action bar
+  private _onRenderActionBar = (): React.ReactNode => {
+    const classNames = getClassNames(this.props.styles!, { theme: this.props.theme! });
+    const step = getCurrentStep(this.props.wizardProps.steps);
```

Comment:
>step [](start = 10, length = 4)

Since this is still getting the steps array.

if in the case of:
click something on the content page, the footer action will change on text, function.
for each content page of this scenario, we will need create a specified footer page, which will be a big work if we have a lot of contents.

instead, I'd suggest move the 
mainAction: IWizardStepAction;

  exitWizardAction: IWizardStepAction;

  backClickAction?: IWizardStepAction;
aslo the title element to the root jsx attribute as the current 'actions' and 'title'

in that case, from the content element, if we need take some other action rather than the default one, the content element can dispatch the state store to tell fabric wizard, what is my actions and titles, fabric take these setting as priority settings to render.

Reply:
Removed the actions.  Lets have the consumer of the control handle it through footer element.